### PR TITLE
[core] Refactor row ID reassignment manifest processing

### PR DESCRIPTION
**Pull request by JingsongLi**
## What changed

- Extract row-ID reassignment planning from `DataEvolutionRowIdReassigner` into a focused planner.
- Add reusable projected `BinaryManifestEntry` and `BinaryDataFileMeta` views.
- Bind projected fields by schema field ID instead of relying on a planner-specific fixed layout.
- Avoid validating or projecting manifest versions in the lightweight binary view.
- Keep the manifest scan hot path on binary strings and serialized rows without materializing POJOs.
- Add dedicated unit tests for both binary views, including arbitrary projections, unsupported fields, null handling, reuse, and lifecycle checks.

## Why

The row-ID reassignment implementation introduced in #8882 combines orchestration, planning state, binary manifest decoding, and compact data structures in one large class. This makes the algorithm difficult to review and gives its inline binary decoding logic a schema that is specific to one caller.

The extracted binary views make projection handling reusable and keep the planner focused on row-ID assignment.

## Impact

This is an internal refactor. Row-ID assignment behavior and manifest serialization remain unchanged. Accessors for projected fields implement the existing `ManifestEntry` and `DataFileMeta` contracts; access to fields that were not projected fails explicitly.

## Validation

```text
mvn -pl paimon-core -DfailIfNoTests=false -DwildcardSuites=none \
  -Dtest=BinaryDataFileMetaTest,BinaryManifestEntryTest,DataEvolutionRowIdReassignerTest test
```

56 tests passed. Checkstyle, Spotless, and Enforcer passed.

### CODE REVIEW

File: paimon-core/src/test/java/org/apache/paimon/io/BinaryDataFileMetaTest.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        assertThat(actual.hasEmbeddedIndex()).isTrue();
+        assertThat(actual.embeddedIndex()).containsExactly(1, 2);
+        assertThat(actual.fileSource()).contains(FileSource.COMPACT);
+        assertThat(actual.valueStatsCols()).containsExactly("value_col");
+        assertThat(actual.hasExternalPath()).isTrue();
+        assertThat(actual.externalPath()).isEqualTo(expected.externalPath());
+        assertThat(actual.externalPathDir()).isEqualTo(expected.externalPathDir());
+        assertThat(actual.hasFirstRowId()).isTrue();
+        assertThat(actual.firstRowId()).isEqualTo(10L);
+        assertThat(actual.firstRowIdValue()).isEqualTo(10L);
```
Comment (leaves12138) [inline paimon-core/src/test/java/org/apache/paimon/io/BinaryDataFileMetaTest.java]: `firstRowIdValue()` was removed from `BinaryDataFileMeta` in the latest commit, but this assertion still calls it, so `paimon-core` fails during test compilation. Please switch this to `actual.nonNullFirstRowId()` (or restore the helper).

**Review by leaves12138 — COMMENTED** [review]
Found one compilation issue in the latest update.

**Review by leaves12138 — APPROVED** [review]
Rechecked the latest changes. The compilation issue is fixed, the manifest field projections remain consistent, and the focused core tests pass.
